src/ClientData/ProcessData.cpp

```diff
@@ -92,11 +92,10 @@ std::optional<ModuleInMemory> ProcessData::FindModuleByPath(const std::string& m
 
 void ProcessData::AddOrUpdateModuleInfo(const ModuleInfo& module_info) {
   absl::MutexLock lock(&mutex_);
-  module_memory_map_.insert_or_assign(
-      module_info.file_path(),
-      ModuleInMemory{module_info.address_start(), module_info.address_end(),
-                     module_info.file_path(), module_info.build_id()});
-  start_addresses_.insert_or_assign(module_info.address_start(), module_info.file_path());
+  ModuleInMemory module_in_memory{module_info.address_start(), module_info.address_end(),
+                                  module_info.file_path(), module_info.build_id()};
+  module_memory_map_.insert_or_assign(module_info.file_path(), module_in_memory);
```

Comment:
Just to clarify: This line is the reason why some functions are now marked as deprecated as they may return only the most recently inserted mapped module, correct?

Nit: If the above is correct, maybe add a brief comment. But if you are addressing this anyway in follow-up PRs, can also be omitted.

Reply:
yes - it is correct.